crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -118,14 +194,10 @@ where
                 let BlockInfo {
                     height,
                     fullness,
-                    block_bytes,
-                    gas_price,
                 } = l2_block;
                 self.inner.update_l2_block_data(
                     height,
                     fullness,
-                    block_bytes,
-                    gas_price,
                 )?;
                 self.metadata_storage
```

Comment:
I don't see why the `set_medatata` should be `async`. Plus since it can return an error, it is better to call before `self.inner.update_l2_block_data`.